core/gsb-api/src/api.rs

```diff
@@ -0,0 +1,936 @@
+use crate::service::StartBuffering;
+use crate::services::{Bind, Find, Services, Unbind};
+use crate::{GsbApiError, WsDisconnect, WsMessagesHandler};
+use actix::Addr;
+use actix_http::ws::{CloseCode, CloseReason};
+use actix_http::StatusCode;
+use actix_web::web::Data;
+use actix_web::Scope;
+use actix_web::{web, HttpRequest, Responder, Result};
+use actix_web_actors::ws::{self};
+use serde::{Deserialize, Serialize};
+use ya_service_api_web::middleware::Identity;
+
+pub(crate) fn web_scope(services: Addr<Services>) -> Scope {
+    actix_web::web::scope(&format!("/{}", crate::GSB_API_PATH))
+        .app_data(Data::new(services))
+        .service(post_services)
+        .service(delete_services)
+        .service(get_service_messages)
+}
+
+#[actix_web::post("/services")]
+async fn post_services(
+    body: web::Json<ServicesBody>,
+    _id: Identity,
+    services: Data<Addr<Services>>,
+) -> Result<impl Responder, GsbApiError> {
+    log::debug!("POST /services Body: {:?}", body);
+    if let Some(listen) = &body.listen {
+        let components = listen.components.clone();
+        let on = listen.on.clone();
+        let bind = Bind {
+            components: components.clone(),
+            addr_prefix: on.clone(),
+        };
+        let response = services.send(bind).await;
+        log::debug!("Service bind result: {:?}", response);
+        response??;
+        let listen_on_encoded = base64::encode(&on);
+        let links = ServicesLinksBody {
+            messages: format!("gsb-api/v1/services/{listen_on_encoded}"),
+        };
+        let services = ServicesBody {
+            listen: Some(ServicesListenBody {
+                on,
+                components,
+                links: Some(links),
+            }),
+        };
+        return Ok(web::Json(services)
+            .customize()
+            .with_status(StatusCode::CREATED));
+    }
+    Err(GsbApiError::BadRequest("Missing listen field".to_string()))
+}
+
+#[actix_web::delete("/services/{key}")]
+async fn delete_services(
+    path: web::Path<ServicesPath>,
+    _id: Identity,
+    services: Data<Addr<Services>>,
+) -> Result<impl Responder, GsbApiError> {
+    let addr = decode_addr(&path.key)?;
+    log::debug!("DELETE service: {}", addr);
+    let unbind = Unbind { addr };
+    let response = services.send(unbind).await;
+    log::debug!("Service delete result: {:?}", response);
+    response??;
+    Ok(web::Json(()))
+}
+
+#[actix_web::get("/services/{key}")]
+async fn get_service_messages(
+    path: web::Path<ServicesPath>,
+    req: HttpRequest,
+    stream: web::Payload,
+    services: Data<Addr<Services>>,
+) -> Result<impl Responder, GsbApiError> {
+    let addr = decode_addr(&path.key)?;
+    log::debug!("GET WS service: {}", addr);
+    let service = services.send(Find { addr }).await??;
+    if let Some(ws_handler) = service.send(StartBuffering).await? {
+        let description = Some("Closing old WS connection on new WS connection".to_string());
```

Comment:
```suggestion
        let description = Some("Closing old WS connection in favor of new WS connection".to_string());
```
I'm not sure how to write this, but first version was not understandable for me

Reply:
Changed.